angular-intro-project/src/app/add-course/add-course.component.ts

```diff
@@ -13,32 +14,46 @@ export class AddCourseComponent implements OnInit {
   courseDate: number;
   courseDuration: number;
   id: number;
+  course: Course;
 
   constructor(private route: ActivatedRoute, private router: Router, private courses: CoursesService) { }
 
   ngOnInit() {
     this.route.params.subscribe(params => {
       this.id = +params.id;
       if (this.id) {
-        const course = this.courses.getItemByID(this.id);
-        this.populateCourseData(course);
-        this.courses.updateCrumbs(this.courseTitle);
+        this.courses.getItemByID(this.id).subscribe((item) => {
```

Comment:
Subscription in subscription is not a common pattern, we use high level observables to handle such cases. I would propose to fix this place in next task. Also we shouldn't forget to unsubscribe.